ci/scripts/backwards-compat-test.sh

```diff
@@ -23,128 +23,23 @@ done
+elif [[ "$profile" != "ci-release" ]]; then
+  echo "Running in ci-release mode"
+  ENABLE_RELEASE_PROFILE=true
+else
     echo "Invalid option: profile must be either ci-dev or ci-release" 1>&2
     exit 1
 fi
 
-################################### ENVIRONMENT VARIABLES
+source backwards-compat-tests/scripts/utils.sh
 
-LOG_DIR=.risingwave/log
-mkdir -p "$LOG_DIR"
+################################### Main
 
-QUERY_LOG_FILE="$LOG_DIR/query.log"
-
-# TODO(kwannoel): automatically derive this by:
-# 1. Fetching major version.
-# 2. Find the earliest minor version of that major version.
-TAG=v0.18.0
-# Duration to wait for recovery (seconds)
-RECOVERY_DURATION=20
-
-################################### TEST UTILIIES
-
-assert_not_empty() {
-  set +e
-  if [[ $(wc -l < "$1" | sed 's/^ *//g') -gt 1 ]]; then
-    echo "assert_not_empty PASSED for $1"
-  else
-    echo "assert_not_empty FAILED for $1"
-    buildkite-agent artifact upload "$1"
-    exit 1
-  fi
-  set -e
-}
-
-assert_eq() {
-  set +e
-  if [[ -z $(diff "$1" "$2") ]]; then
-    echo "assert_eq PASSED for $1 and $2"
-  else
-    echo "FAILED"
-    buildkite-agent artifact upload "$1"
-    buildkite-agent artifact upload "$2"
-    exit 1
-  fi
-  set -e
-}
-
-################################### QUERIES
-
-run_sql () {
-    psql -h localhost -p 4566 -d dev -U root -c "$@"
-}
-
-seed_table() {
-  START="$1"
-  END="$2"
-  for i in $(seq "$START" "$END")
-  do
-    run_sql "INSERT into t values ($i, $i);" 1>$QUERY_LOG_FILE 2>&1
-  done
-  run_sql "flush;"
-}
-
-random_delete() {
-  START=$1
-  END=$2
-  COUNT=$3
-  for i in $(seq 1 "$COUNT")
-  do
-    run_sql "DELETE FROM t WHERE v1 = $(("$RANDOM" % END));" 1>$QUERY_LOG_FILE 2>&1
-  done
-  run_sql "flush;"
-}
-
-random_update() {
-  START=$1
-  END=$2
-  COUNT=$3
-  for _i in $(seq 1 "$COUNT")
-  do
-    run_sql "UPDATE t SET v2 = v2 + 1 WHERE v1 = $(("$RANDOM" % END));" 1>$QUERY_LOG_FILE 2>&1
-  done
-  run_sql "flush;"
-}
-
-# Setup table and materialized view.
-# Run updates and deletes on the table.
-# Get the results.
-# TODO: Run nexmark, tpch queries
-run_sql_old_cluster() {
-  run_sql "CREATE TABLE t(v1 int primary key, v2 int);"
-
-  seed_table 1 10000
-
-  run_sql "CREATE MATERIALIZED VIEW m as SELECT * from t;" &
-  CREATE_MV_PID=$!
-
-  seed_table 10001 20000
-
-  random_update 1 20000 1000
-
-  random_delete 1 20000 1000
-
-  wait $CREATE_MV_PID
-
-  run_sql "CREATE MATERIALIZED VIEW m2 as SELECT v1, sum(v2) FROM m GROUP BY v1;"
-
-  run_sql "select * from m ORDER BY v1;" > BEFORE_1
-  run_sql "select * from m2 ORDER BY v1;" > BEFORE_2
-}
-
-# Just check if the results are the same as old cluster.
-run_sql_new_cluster() {
-  run_sql "SELECT * from m ORDER BY v1;" > AFTER_1
-  run_sql "select * from m2 ORDER BY v1;" > AFTER_2
-}
-
-run_updates_and_deletes_new_cluster() {
-  random_update 1 20000 1000
-  random_delete 1 20000 1000
-}
-
-################################### CLUSTER CONFIGURATION
+OLD_TAG=1.0.0
```

Comment:
can these two variables be changed by passing the environment variable through Buildkite?

Reply:
or default to the latest tag?
list all the tags in the Repo, and sort them in some magical way then determine

Reply:
Updated both to be automatically derived. We run backwards compat test for current version against N versions before it. See https://github.com/risingwavelabs/risingwave/blob/c180eb8ddbffd058de454aae042c9d387391a16f/backwards-compat-tests/scripts/utils.sh#L124